test/integration/api/app-users.js

```diff
@@ -4,10 +4,10 @@ const testData = require('../../data/xml');
 
 describe('api: /projects/:id/app-users', () => {
   describe('POST', () => {
-    it('should return 403 unless the user is allowed to create', testService((service) =>
+    it('should return 403 if unauthenticated', testService((service) =>
       service.post('/v1/projects/1/app-users')
         .send({ displayName: 'test1' })
-        .expect(403)));
+        .expect(401)));
```

Comment:
Would there be value in a corresponding test case for authenticated users who are unauthorised?

Reply:
Similar question from me. I'd expect every endpoint that calls `auth.canOrReject()` to have a test asserting a 403. This test now never reaches the actual endpoint logic: it's just being stopped by the preprocessor, which we test elsewhere.